src/providers/wfs/qgswfsrequest.cpp

```diff
@@ -201,20 +201,20 @@ bool QgsWfsRequest::sendGET( const QUrl &url, bool synchronous, bool forceRefres
       // he needs the authentication to run.
       // The processEvents() call gives the auth manager the chance to show a dialog and
       // once done with that, we can wake the downloaderThread again and continue the download.
-      if ( !downloaderThread->isFinished() )
+      if ( !threadFinished )
       {
         QgsApplication::instance()->processEvents();
         waitConditionMutex.lock();
         waitCondition.wakeAll();
         waitConditionMutex.unlock();
       }
     }
-
-    success = downloaderThread->success();
+    while ( !downloaderThread->isFinished() )
```

Comment:
busy loop that will burn CPU. Perhaps add some tiny sleep amount in it ?
Or perhaps even better: downloaderThread->wait() ?